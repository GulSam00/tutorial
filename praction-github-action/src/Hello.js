import React from "react";

const Hello = ({ first, name }) => {
  return (
    <div>
      <b>{first}</b>&nbsp;
      <span>{name}</span>
    </div>
  );
};

export default Hello;
