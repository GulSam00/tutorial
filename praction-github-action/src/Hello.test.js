import React from "react";
import { render } from "@testing-library/react";
import Hello from "./Hello";

describe("<Hello />", () => {
  it("matches snapshot", () => {
    const utils = render(<Hello first="K" name="Kim" />);
    expect(utils.container).toMatchSnapshot();
  });
  it("shows the props correctly", () => {
    const utils = render(<Hello first="K" name="Kim" />);
    utils.getByText("K"); // velopert 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText("Kim"); // (김민준) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
  });
});
