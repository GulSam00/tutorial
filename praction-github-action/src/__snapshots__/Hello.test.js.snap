// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      K
    </b>
    Â 
    <span>
      Kim
    </span>
  </div>
</div>
`;
